qurator/eynollah/processor.py

```diff
@@ -0,0 +1,59 @@
+from json import loads
+from pkg_resources import resource_string
+from tempfile import NamedTemporaryFile
+from os.path import join
+
+from ocrd import Processor
+from ocrd_modelfactory import page_from_file
+from ocrd_models.ocrd_page import to_xml
+from ocrd_utils import (
+    getLogger,
+    MIMETYPE_PAGE,
+    assert_file_grp_cardinality,
+    make_file_id
+)
+
+from .eynollah import Eynollah
+from .utils.pil_cv2 import pil2cv
+
+OCRD_TOOL = loads(resource_string(__name__, 'ocrd-tool.json').decode('utf8'))
+
+class EynollahProcessor(Processor):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools']['ocrd-eynollah-segment']
+        kwargs['version'] = OCRD_TOOL['version']
+        super().__init__(*args, **kwargs)
+
+    def process(self):
+        LOG = getLogger('eynollah')
+        assert_file_grp_cardinality(self.input_file_grp, 1)
+        assert_file_grp_cardinality(self.output_file_grp, 1)
+        for n, input_file in enumerate(self.input_files):
+            page_id = input_file.pageId or input_file.ID
+            LOG.info("INPUT FILE %s (%d/%d) ", page_id, n + 1, len(self.input_files))
+            pcgts = page_from_file(self.workspace.download_file(input_file))
+            self.add_metadata(pcgts)
+            page = pcgts.get_Page()
+            page_image, _, _ = self.workspace.image_from_page(page, page_id, feature_filter='binarized')
```

Comment:
Unfortunately, due to [Pillow deficiencies](https://github.com/OCR-D/core/blob/d2c6216f351b3c9af6f4f267bc6332de1482e7bf/ocrd/ocrd/workspace.py#L402-L435), that still does not guarantee we end up with an image that includes all metadata:

```suggestion
            page_image = self.workspace.resolve_image_as_pil(page.imageFilename)
```

But maybe you could pass the `OcrdExif` into `Eynollah` explicitly and use `self.workspace.resolve_image_exif(page.imageFilename)` here?

Reply:
Lest I forget: It would be nice to get the image along with the exif information:

```
exif, img_pil = self.wrokspace.resolve_image_exif(page.imageFilename)
```

since we're opening the image anyway.
```

Reply:
Yes, but that would be a backwards-imcompatible change of the API. How about de-deprecating `resolve_image_pil` and adding another kwarg `exif=False` which controls whether the return value should become a tuple with the OcrdExif instance as secondary value?

Reply:
True. I opened https://github.com/OCR-D/core/issues/685 for this so we can revisit once we get to the API changes in core.